### ISSUE TITLE
Self import with TypeVar fallback results in unknown errors

### ISSUE THREAD

**Issue by cpburnz** [microsoft/pyright#11223]
**Describe the bug**

[typing.Self](https://docs.python.org/3/library/typing.html#typing.Self) was added in Python 3.11. Its documentation says it's equivalent to `TypeVar("Self", bound="Foo")`, which can be used in Python<3.11.

If you try to use a `Self` import with a `TypeVar` fallback, *pyright* produces multiple errors about unknown. Example:

```python
try:
	from typing import Self
except ImportError:
	from typing import TypeVar
	Self = TypeVar("Self", bound='Base')

class Base(object):
	@classmethod
	def make(cls: type[Self]) -> Self:  # Line 9
		return cls()

class Sub(Base):
	pass

sub: Sub = Sub.make()  # Line 15
```

This produces the following errors (with typeCheckingMode="strict"):

```
/home/caleb/test/example1.py
  /home/caleb/test/example1.py:9:6 - error: Return type is unknown (reportUnknownParameterType)
  /home/caleb/test/example1.py:9:11 - error: Type of parameter "cls" is partially unknown
    Parameter type is "type[Unknown]" (reportUnknownParameterType)
  /home/caleb/test/example1.py:9:21 - error: Variable not allowed in type expression (reportInvalidTypeForm)
  /home/caleb/test/example1.py:9:31 - error: Variable not allowed in type expression (reportInvalidTypeForm)
  /home/caleb/test/example1.py:15:1 - error: Type of "sub" is unknown (reportUnknownVariableType)
  /home/caleb/test/example1.py:15:12 - error: Type of "make" is partially unknown
    Type of "make" is "() -> Unknown" (reportUnknownMemberType)
6 errors, 0 warnings, 0 informations
```

However, if I use either form independently, no errors are produced:

```python
from typing import Self

class Base(object):
	@classmethod
	def make(cls: type[Self]) -> Self:
		return cls()

class Sub(Base):
	pass

sub: Sub = Sub.make()
```

```python
from typing import TypeVar
Self = TypeVar("Self", bound='Base')

class Base(object):
	@classmethod
	def make(cls: type[Self]) -> Self:
		return cls()

class Sub(Base):
	pass

sub: Sub = Sub.make()
```

This was tested using Python 3.13.11 and pyright 1.1.408.

**Comment by erictraut** [comment]
Pyright is working as intended here.

The symbol `Self` in your example cannot be used in a type expression because it has multiple incompatible definitions.

The recommended way to use `Self` in a way that's compatible with Python < 3.11 is to import it from `typing_extensions`.

```python
from typing_extensions import Self
```

If you don't want to import from `typing_extensions` for some reason, you will need to specify which definition you want pyright to use for static type analysis. 

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import Self
else:
    try:
        from typing import Self
    except ImportError:
        from typing import TypeVar

        Self = TypeVar("Self", bound="Base")
```
